SIT.Manager.Avalonia/Services/AkiServerService.cs

```diff
@@ -0,0 +1,103 @@
+
+        public string ServerFilePath => !string.IsNullOrEmpty(ServerDirectory) ? Path.Combine(ServerDirectory, SERVER_EXE) : string.Empty;
+
+        public RunningState State { get; private set; } = RunningState.NotRunning;
+
+        public event EventHandler<DataReceivedEventArgs>? OutputDataReceived;
+        public event EventHandler<RunningState>? RunningStateChanged;
+
+        public AkiServerService(IManagerConfigService configService) {
+            _configService = configService;
+        }
+
+        private void ExitedEvent(object? sender, EventArgs e) {
+            if (State == RunningState.Running && !_stopRequest) {
+                State = RunningState.StoppedUnexpectedly;
+            }
+            else {
+                State = RunningState.NotRunning;
+            }
+
+            _stopRequest = false;
+            RunningStateChanged?.Invoke(this, State);
+        }
+
+        public bool IsUnhandledInstanceRunning() {
+            Process[] akiServerProcesses = Process.GetProcessesByName(SERVER_EXE.Replace(".exe", ""));
+
+            if (akiServerProcesses.Length > 0) {
+                if (_serverProcess == null || _serverProcess.HasExited) {
+                    return true;
+                }
+
+                foreach (Process akiServerProcess in akiServerProcesses) {
+                    if (_serverProcess.Id != akiServerProcess.Id) {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        public void Start() {
+            if (State == RunningState.Running) {
+                return;
+            }
+
+            _serverProcess = new Process();
+
+            _serverProcess.StartInfo.FileName = ServerFilePath;
+            _serverProcess.StartInfo.WorkingDirectory = ServerDirectory;
+            _serverProcess.StartInfo.UseShellExecute = false;
+            _serverProcess.StartInfo.StandardOutputEncoding = Encoding.UTF8;
+            _serverProcess.StartInfo.RedirectStandardOutput = true;
+            _serverProcess.StartInfo.CreateNoWindow = true;
+            _serverProcess.EnableRaisingEvents = true;
+            _serverProcess.OutputDataReceived += new DataReceivedEventHandler((sender, e) => OutputDataReceived?.Invoke(sender, e));
+            _serverProcess.Exited += new EventHandler((sender, e) => ExitedEvent(sender, e));
+
+            _serverProcess.Start();
+            _serverProcess.BeginOutputReadLine();
+
+            State = RunningState.Running;
+
+            RunningStateChanged?.Invoke(this, State);
+        }
+
+        public void Stop() {
+            if (State == RunningState.NotRunning || _serverProcess == null || _serverProcess.HasExited) {
+                return;
+            }
+
+            _stopRequest = true;
+
+            // Kill the server process
+            _serverProcess.Kill();
```

Comment:
I'm not sure if it matters, at least for the case of AKI server but the recommended way to kill processes is by closing their main window with `#Process.CloseMainWindow()`.
something like
```suggestion
            bool clsMsgSent = _serverProcess.CloseMainWindow();
            if (clsMsgSent)
                        serverProcess.WaitForExit();
            else
			serverProcess.Kill();
```

Reply:
I did try something similar to thig before but it didn;t work, I imagine it's because we never actually launch a main window which we can actually close. This left the only option I could think of as the rather harsh kill method, I'll put it in anyway as it can't hurt to try.

Reply:
You make a good point there, I didn't think about that fact that we never open a window when launching via the launcher. This would be a moot point then